pocs/camera/camera.py

```diff
@@ -634,47 +698,54 @@ def _process_fits(self, file_path, info):
         of the appropriate subcomponent class, or from a dictionary of keyword arguments for the
         subcomponent class' constructor.
 
         Args:
-            subcomponent (instance of class_name | dict): the subcomponent object, or the keyword
+            subcomponent (instance of sub_name | dict): the subcomponent object, or the keyword
                 arguments required to create it.
-            sub_name (str): name of the subcomponent, e.g. 'focuser'. Will be used as the attribute
-                name, and must also match the name corresponding POCS submodule for this
-                subcomponent, e.g. `pocs.focuser`
-            class_name (str): name of the subcomponent class, e.g. 'Focuser'
-            base_class (class): the base class for the subcomponent, e.g.
-                `pocs.focuser.AbtractFocuser`, used to check whether subcomponent is an instance.
+            class_name (str): name of the subcomponent class, e.g. 'Focuser'. Lower cased version
+                will be used as the attribute name, and must also match the name of the
+                corresponding POCS submodule for this subcomponent, e.g. `pocs.focuser`.
         """
         if subcomponent:
+            base_module_name = "pocs.{}.{}".format(class_name.casefold(), class_name.casefold())
+            try:
+                base_module = load_module(base_module_name)
+            except AttributeError as err:
+                self.logger.critical("Couldn't import {} base class module {}!".format(
+                    class_name, base_module_name))
+                raise err
+            base_class = getattr(base_module, "Abstract{}".format(class_name))
+
             if isinstance(subcomponent, base_class):
                 self.logger.debug("{} received: {}".format(class_name, subcomponent))
-                setattr(self, sub_name, subcomponent)
-                getattr(self, sub_name).camera = self
+                setattr(self, class_name.casefold(), subcomponent)
+                getattr(self, class_name.casefold()).camera = self
             elif isinstance(subcomponent, dict):
-                module_name = 'pocs.{}.{}'.format(sub_name, subcomponent['model'])
+                module_name = 'pocs.{}.{}'.format(class_name.casefold(), subcomponent['model'])
                 try:
                     module = load_module(module_name)
                 except AttributeError as err:
                     self.logger.critical("Couldn't import {} module {}!".format(
                         class_name, module_name))
                     raise err
-                else:
-                    subcomponent_kwargs = copy.copy(subcomponent)
-                    subcomponent_kwargs.update({'camera': self, 'config': self.config})
-                    setattr(self, sub_name, getattr(module, class_name)(**subcomponent_kwargs))
+                subcomponent_kwargs = copy.copy(subcomponent)
```

Comment:
Do you want a `deepcopy` here? If the `subcomponent` contains any lists or other types within it we would probably want to. It's a small enough object that it probably doesn't hurt.

Reply:
Yes, I should be using `deepcopy`. `subcomponent` will generally contain lists, etc.